StonehearthEditor/MainForm.cs

```diff
@@ -108,36 +101,82 @@ private void netWorthVisualizerToolStripMenuItem_Click(object sender, EventArgs
                 mNetWorthVisualizer.UpdateNetWorthData();
                 return;
             }
+
             mNetWorthVisualizer = new NetWorthVisualizer();
             mNetWorthVisualizer.SetManifestView(manifestView);
-            //mNetWorthVisualizer.UpdateNetWorthData();
+            ////mNetWorthVisualizer.UpdateNetWorthData();
             mNetWorthVisualizer.Show(this);
         }
 
         private void chooseModDirectory()
         {
-            DialogResult result = modsFolderBrowserDialog.ShowDialog();
+            var dialog = new FolderSelectDialog()
+            {
+                DirectoryPath = kModsDirectoryPath ?? Environment.CurrentDirectory,
+                Title = "Stonehearth Mods Root Directory"
+            };
 
-            if (result == DialogResult.OK)
+            DialogResult result = DialogResult.Abort;
+
+            do
             {
-                string newPath = modsFolderBrowserDialog.SelectedPath;
-                if (!string.IsNullOrWhiteSpace(newPath))
+                result = dialog.ShowDialog();
+
+                if (result == DialogResult.OK)
                 {
-                    kModsDirectoryPath = JsonHelper.NormalizeSystemPath(modsFolderBrowserDialog.SelectedPath);
-                    Properties.Settings.Default["ModsDirectory"] = kModsDirectoryPath;
-                    Properties.Settings.Default.Save();
+                    string newPath = dialog.DirectoryPath;
+
+                    if (this.checkModsFolder(ref newPath))
+                    {
+                        kModsDirectoryPath = newPath;
```

Comment:
Done: https://github.com/stonehearth/stonehearth-editor/pull/1/files#diff-7c4ad0a72c371a4ccd964b2e2bc2b0cbR131

Because `checkModsFolder` might modify the path (by attaching `mods/` where appropriate), the normalizing has to happen when assigning it to `kModsDirectory`.